project_forecast_line/tests/test_forecast_line.py

```diff
@@ -603,10 +603,155 @@ def test_task_forecast_lines_consolidated_forecast_overallocation_multiple_tasks
+                {
+                    "name": "TaskDiffRoles",
+                    "project_id": project.id,
+                    "forecast_role_id": self.role_consultant.id,
+                    "forecast_date_planned_start": "2022-02-14",
+                    "forecast_date_planned_end": "2022-02-14",
+                }
+            )
+            task.remaining_hours = 8
+            task.user_ids = self.user_consultant
+            task_forecast = self.env["forecast.line"].search(
+                [("task_id", "=", task.id)]
+            )
+            self.assertEqual(len(task_forecast), 1)
+            # using assertEqual on purpose here
+            self.assertEqual(task_forecast.forecast_hours, -8.0)
+            self.assertEqual(task_forecast.consolidated_forecast, 1.0)
+            employee_forecast = self.env["forecast.line"].search(
+                [("employee_id", "=", self.employee_consultant.id)]
+            )
+            # we can take first line to check as forecast values are equal
+            forecast_consultant = employee_forecast.filtered(
+                lambda l: l.res_model == "hr.employee.forecast.role"
+                and l.forecast_role_id == self.role_consultant
+            )[0]
+            self.assertEqual(forecast_consultant.forecast_hours, 6.0)
+            self.assertAlmostEqual(forecast_consultant.consolidated_forecast, -0.25)
+            forecast_pm = employee_forecast.filtered(
+                lambda l: l.res_model == "hr.employee.forecast.role"
+                and l.forecast_role_id == self.role_pm
+            )[0]
+            self.assertEqual(forecast_pm.forecast_hours, 2.0)
+            self.assertAlmostEqual(forecast_pm.consolidated_forecast, 0.25)
+
+    def test_task_forecast_lines_employee_main_role(self):
+        """
+        Test forecast lines when employee has different roles
+        and different from employee's role is assigned to the task.
+
+        Employee has 2 forecast_role_id: consultant 75% and project manager 25%,
+        working 8h per day (standard calendar).
+        Create a task with forecast role developer, with remaining time = 8h
+        and a scheduled period starting and ending on the same day (22/01/2022 for instance).
+        Assign this task to the user.
+
+        Expected: for the user, on 22/01/2022, 3 forecast lines.
+        """
+        self.env["hr.employee.forecast.role"].create(
+            {
+                "employee_id": self.employee_consultant.id,
+                "role_id": self.role_pm.id,
+                "date_start": "2022-01-01",
+                "rate": 25,
+                "sequence": 1,
+            }
+        )
+        consultant_role = self.env["hr.employee.forecast.role"].search(
+            [
+                ("employee_id", "=", self.employee_consultant.id),
+                ("role_id", "=", self.role_consultant.id),
+            ]
+        )
+        consultant_role.rate = 75
```

Comment:
I would move this inside the with freeze_time() block below. If you don't do this you will have the forecast lines for the employee created relative to the test run date, and likely there will be no line for the 14/01/2022 which is officially in the past as of now.